controllers/rack.go

```diff
@@ -86,14 +86,15 @@ func (r *SingleClusterReconciler) reconcileRacks() reconcileResult {
 
 		failedPods, _ := getFailedAndActivePods(podList)
 		if len(failedPods) != 0 {
-			r.Log.Info("Upgrading or rolling restart rack with failed pods", "rackID", state.Rack.ID, "failedPods", failedPods)
+			r.Log.Info("Reconcile the failed pods in the Rack", "rackID", state.Rack.ID, "failedPods", failedPods)
 
-			_, res = r.upgradeOrRollingRestartRack(found, state, ignorablePods, failedPods)
-			if !res.isSuccess {
+			if res = r.reconcileRack(
```

Comment:
Another approach:
- Here we are handling failed pods only, so can we fetch the scaled down pod names in failed pods and scale down those in a single go.
- reconcileRack will do scale-up also, should we do scale-up operation in failed pods loop or in next loop.

Reply:
- Does it any much value? In case of scale down of failed pod, we don't wait anywhere so even the current flow should be quick.
- Thinking about scale-up... Not decided

Reply:
we may not be waiting before downsizing the Sts in `scaleDownRack ` but [here](https://github.com/aerospike/aerospike-kubernetes-operator/blob/8593a827e046b934d4c8dbc3a238275e63b9a279/controllers/rack.go#L798) we are waiting for the Sts to be ready after each scale-down, which might be time taking.

Reply:
Now we don't wait.